tests/chainerx_tests/unit_tests/routines_tests/test_logic.py

```diff
@@ -205,3 +205,77 @@ def forward_xp(self, inputs, xp):
         a, = inputs
         b = xp.logical_not(a)
         return b,
+
+
+def compute_all(xp, a, axis, keepdims):
+    return xp.all(a, axis=axis, keepdims=keepdims)
+
+
+def compute_any(xp, a, axis, keepdims):
+    return xp.any(a, axis=axis, keepdims=keepdims)
+
+
+@chainerx.testing.numpy_chainerx_array_equal()
+@pytest.mark.parametrize('keepdims', [False, True])
+@pytest.mark.parametrize('shape,axis', [
+    ((), None),
+    ((), ()),
+    ((2,), None),
+    ((2,), ()),
+    ((2,), 0),
+    ((2,), (0,)),
+    ((2,), (-1,)),
+    ((2, 3), None),
+    ((2, 3), ()),
+    ((2, 3), 0),
+    ((2, 3), (0,)),
+    ((2, 3), (1,)),
+    ((2, 3), (-1,)),
+    ((2, 3), (-2,)),
+    ((2, 3), (0, 1)),
+    ((2, 3), (-2, -1)),
+    ((1, 3), None),  # Reduce over 1-dim axis
+    ((0, 3), None),  # Reduce over 0-dim axis
+    # Reduce over axes that are in the middle or apart
+    ((2, 3, 4), (1,)),
+    ((2, 3, 4), (0, 2)),
+    # Reduce over axes that are apart and/or unsorted
+    ((2, 3), (1, 0)),
+    ((2, 3, 4), (2, 0)),
+    ((2, 3, 4), (2, 0, 1)),
+    ((2, 3, 4), (-2, 2, 0)),
+])
+@pytest.mark.parametrize_device(['native:0', 'cuda:0'])
+@pytest.mark.parametrize('func', [
+    compute_all,
+    compute_any
+])
+def test_logical_reductions(func, xp, device, shape, axis, keepdims, dtype):
+    a = array_utils.create_dummy_ndarray(xp, shape, dtype)
```

Comment:
`array_utils.create_dummy_array` is not appropriate for these tests, because the output of `all` will be always `False` if `size > 0`.
https://github.com/chainer/chainer/blob/master/tests/chainerx_tests/array_utils.py#L43

Reply:
Have updated the test generation. Please review.